react/types.tsx

```diff
@@ -1,73 +1,21 @@
-import { Address } from 'vtex.checkout-graphql'
-
-export interface LineFragment {
-  name: keyof Fields
-  delimiter?: string
-  delimiterAfter?: string
-}
+import type { Address } from 'vtex.checkout-graphql'
+import type {
+  AddressFieldsData,
+  DisplayData,
+  LocationSelectData,
+  Field,
+} from 'vtex.country-data-settings'
 
 export type AddressFields = keyof Omit<Address, '__typename' | 'geoCoordinates'>
 
-export interface PostalCodeData {
-  /**
-   * Link to the country postal code service to help the user
-   * retrieve the postal code for their address.
-   */
-  forgottenURL?: string
-}
-
-export interface Field {
-  label: string
-  name?: AddressFields
-  hidden?: boolean
-  maxLength?: number
-  size?: string
-  required?: boolean
-  autoComplete?: string
-  optionsCaption?: string
-  options?: Array<{ label: string; value: string }>
-  elementName?: string
-  /**
-   * Optional mask to format the field for display.
-   *
-   * E.g.: the mask "99999-999" for Brazil postal code, it would
-   * format the value "22250040" to "22250-040".
-   */
-  mask?: string
-  additionalData?: PostalCodeData | null
-  pattern?: string
-}
-
-export interface Display {
-  minimal: LineFragment[][]
-  compact: LineFragment[][]
-  extended: LineFragment[][]
-}
-
-export interface Fields {
-  country?: Field
-  street?: Field
-  number?: Field
-  complement?: Field
-  reference?: Field
-  neighborhood?: Field
-  state?: Field
-  city?: Field
-  receiverName?: Field
-  postalCode?: Field
-}
-
-export interface LocationSelect {
-  countryData: object
-  fields: Field[]
-}
-
 export interface CountryRules {
-  fields: Fields
-  display: Display
-  locationSelect?: LocationSelect
+  fields: AddressFieldsData
+  display: DisplayData
+  locationSelect?: LocationSelectData | null
```

Comment:
shouldn't the null be defined on the original LocationSelect type on `vtex.country-data-settings`?

Reply:
no, since this type is derived from the graphql definition, so we need to include it here